services/debug/service.go

```diff
@@ -0,0 +1,72 @@
+package debug
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+)
+
+// Make sure that Service implements node.Service interface.
+var _ node.Service = (*Service)(nil)
+
+// WhisperService whisper interface to add key pairs
+type WhisperService interface {
+	SubscribeEnvelopeEvents(events chan<- whisper.EnvelopeEvent) event.Subscription
+}
+
+// Poster interface for a posting shh messages.
+type Poster interface {
+	Post(context.Context, whisper.NewMessage) (hexutil.Bytes, error)
+}
+
+// Service represents provides some debugging specific endpoints.
+type Service struct {
+	w WhisperService
+	p Poster // *shhext.PublicAPI
+}
+
+// New returns a new Service.
+func New(w WhisperService) *Service {
+	return &Service{w: w}
+}
+
+// Protocols returns a new protocols list. In this case, there are none.
+func (s *Service) Protocols() []p2p.Protocol {
+	return []p2p.Protocol{}
+}
+
+// APIs returns a list of new APIs.
+func (s *Service) APIs() []rpc.API {
+
+	return []rpc.API{
+		{
+			Namespace: "debug",
+			Version:   "1.0",
+			Service:   NewAPI(s),
```

Comment:
another option is to add this api next to sshext api. so that we avoid creating more services just for apis.

Reply:
That sounds good. I forgot that a service does not define a namespace but APIs belong to service do that. I think it's worth to explore and move this API to "shhext" service but keep `Namespace: "debug"`.

Reply:
ATM the debug service is being registered based on a specific config option. Even i have multiple apis assigned to a service, I'm not sure how can we're going to be able to disable/enable just some them.